### PR TITLE
New icons and style homogeneization

### PR DESCRIPTION
**Pull request by frankrousseau**
**Problem**

* Some icons are not self explanatory
* The styles are not homegeneous

**Solution**

* Use dedicated icons through the Kitsu Icon widget
* Define colors and schemes for selectable elements
* Define colors for highlightable elements
* Make all button as rounded squares

+ fixes:

* Preview right-click scrubbing was not working anymore
* Clicking on the same tasks in notifications and news moved the player timeline

### CODE REVIEW

File: src/components/pages/Playlist.vue [hunk 1]
```diff
@@ -1514,13 +1536,14 @@ export default {
...
-    transform: scale(1.03);
-    border-right: 2px solid $dark-purple;
+    transform: scale(1.02);
+    border: 2px solid var(--background-selectable);
   }
 }
 
 .playlist-item.selected {
-  border-right: 3px solid $light-green;
+  border: 2px solid var(--background-selected);
```
Comment (NicoPennec) [inline src/components/pages/Playlist.vue:1545]: Prefer to use `border-color`

File: src/components/pages/ProductionNewsFeed.vue [hunk 2]
```diff
@@ -442,6 +428,17 @@ export default {
       'user'
     ]),
 
+    statMax() {
+      if (this.newsStats) {
+        return Object.keys(this.newsStats).reduce((max, stat) => {
+          console.log(stat, this.newsStats[stat], max)
```
Comment (NicoPennec) [inline src/components/pages/ProductionNewsFeed.vue:434]: Log to remove.

File: src/components/previews/VideoViewer.vue [hunk 3]
```diff
@@ -397,9 +399,15 @@ export default {
...
       }
+      if (currentTimeRaw === 0) {
+        return 0
+      }
       let frame = Math.ceil(currentTimeRaw / this.frameDuration) + 1
+      console.log('frame bf', frame)
       frame = Number(frame.toPrecision(4))
+      console.log('frame bf2', frame)
       frame = Math.min(frame, this.nbFrames)
+      console.log('frame after', frame)
```
Comment (NicoPennec) [inline src/components/previews/VideoViewer.vue:410]: Useful logs?

File: src/components/pages/ProductionNewsFeed.vue [hunk 4]
```diff
@@ -956,17 +933,20 @@ export default {
...
   cursor: pointer;
-  padding-top: 0.5em;
-  padding-bottom: 0.5em;
+  margin-bottom: 1px;
+  padding-top: 0.3em;
+  padding-bottom: 0.3em;
   border-radius: 0.5em;
+  transition: all 0.1s linear;
 
   &:hover {
```
Comment (NicoPennec) [inline src/components/pages/ProductionNewsFeed.vue:944]: Prefer to use `border-color`.

File: src/components/pages/TeamSchedule.vue [hunk 5]
```diff
@@ -55,9 +55,12 @@
             v-model="selectedPerson"
           />
         </div>
+        <div class="filler"></div>
+        <div class="button" @click="$refs.schedule.scrollToToday()">today</div>
       </div>
 
       <schedule
+        ref="schedule"
```
Comment (NicoPennec) [inline src/components/pages/TeamSchedule.vue:63]: This `ref` is never used.

File: src/components/pages/entities/EntityOutputFiles.vue [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+    ...mapActions(['getEntityPreviewFiles']),
+
+    getTaskType(outputFile) {
+      const task = this.taskMap.get(outputFile.task_id)
+      return this.taskTypeMap.get(task.task_type_id)
+    },
+
+    getDownloadPath(outputFileId) {
+      const type = this.isMovie ? 'movies' : 'pictures'
+      return `/api/${type}/originals/output-files/` + `${outputFileId}/download`
```
Comment (NicoPennec) [inline src/components/pages/entities/EntityOutputFiles.vue:154]: Use a template string down the line:
```suggestion
      return `/api/${type}/originals/output-files/${outputFileId}/download`
```

File: src/components/pages/playlists/PlaylistPlayer.vue [hunk 7]
```diff
@@ -542,9 +540,9 @@
           :class="{
             'comparison-button': true,
             'flexrow-item': true,
-            'playlist-button': true,
-            active: isComparing
+            'playlist-button': true
           }"
```
Comment (NicoPennec) [inline src/components/pages/playlists/PlaylistPlayer.vue:544]: Use a native `class` directly, instead of a directive.

File: src/components/pages/playlists/PlaylistPlayer.vue [hunk 8]
```diff
@@ -664,17 +662,17 @@
           v-if="(isCurrentUserManager || isCurrentUserSupervisor) && tempMode"
         ></div>
         <button-simple
-          @click="isAnnotationsDisplayed = !isAnnotationsDisplayed"
           :class="{
             'playlist-button': true,
-            'flexrow-item': true,
-            active: isAnnotationsDisplayed
+            'flexrow-item': true
           }"
```
Comment (NicoPennec) [inline src/components/pages/playlists/PlaylistPlayer.vue:668]: idem, use `class`

File: src/components/pages/playlists/PlaylistPlayer.vue [hunk 9]
```diff
@@ -689,9 +687,9 @@
         <button-simple
           :class="{
             'playlist-button': true,
-            'flexrow-item': true,
-            active: isTyping
+            'flexrow-item': true
           }"
```
Comment (NicoPennec) [inline src/components/pages/playlists/PlaylistPlayer.vue:691]: idem, use `class`

File: src/components/pages/playlists/PlaylistPlayer.vue [hunk 10]
```diff
@@ -729,9 +727,9 @@
           @click="isLaserModeOn = !isLaserModeOn"
           :class="{
             'playlist-button': true,
```
Comment (NicoPennec) [inline src/components/pages/playlists/PlaylistPlayer.vue:729]: idem, use `class`

File: src/components/previews/PreviewPlayer.vue [hunk 11]
```diff
@@ -1589,9 +1554,11 @@ export default {
...
         const annotations = []
         if (this.currentPreview.annotations.forEach) {
-          this.currentPreview.annotations.forEach(a =>
-            annotations.push({ ...a })
-          )
+          this.currentPreview.annotations.forEach(a => {
+            if (a.time >= 0) {
+              annotations.push({ ...a })
+            }
+          })
```
Comment (NicoPennec) [inline src/components/previews/PreviewPlayer.vue:1561]: You can use an `Array.filter()` here:

```js
const annotations = this.currentPreview.annotations.filter(a => a.time >= 0);
```

File: src/components/sides/TaskInfo.vue [hunk 12]
```diff
@@ -8,7 +8,7 @@
     ></div>
     <div class="side task-info">
       <action-panel
-        v-if="withActions && (!isConceptTask || selectedConcepts.size > 0)"
+        v-if="extendable && (!isConceptTask || selectedConcepts.size > 0)"
```
Comment (NicoPennec) [inline src/components/sides/TaskInfo.vue:11]: Why this change?
The property `withActions` is designed to hide the action panel.
The property `extendable` is designed to allow to resize the parent panel.
Their behaviors are not linked.

File: src/components/sides/TaskInfo.vue [hunk 13]
```diff
@@ -1653,7 +1653,14 @@ export default {
...
-  height: 100%;
+  position: relative;
+  bottom: 0;
+  right: 0;
+  left: 0;
+}
+
+.no-selection-separator {
+  background-color: var(--border-alt);
+  margin: 1em 1em;
```
Comment (NicoPennec) [inline src/components/sides/TaskInfo.vue:1664]: shortcut notation:
```suggestion
  margin: 1em;
```

File: src/components/tops/ActionPanel.vue [hunk 14]
```diff
@@ -1830,29 +1823,22 @@ div.assignation {
...
-
   &.active {
-    border: 2px solid $light-green;
-    color: $light-green;
-
-    &:hover {
-      border: 2px solid $light-green;
-      color: $light-green;
-    }
+    border: 2px solid var(--text-selected);
```
Comment (NicoPennec) [inline src/components/tops/ActionPanel.vue:1840]: Prefer to use `border-color`

File: src/components/widgets/SortingInfo.vue [hunk 15]
```diff
@@ -1,7 +1,8 @@
 <template functional>
-  <div class="has-text-right" v-if="props.sorting.length > 0">
+  <div class="" v-if="props.sorting.length > 0">
+    <div @click="listeners['clear-sorting']()" class="sort-tag">
     {{ props.label }} {{ props.sorting[0].name }}
```
Comment (NicoPennec) [inline src/components/widgets/SortingInfo.vue:4]: indent:
```suggestion
      {{ props.label }} {{ props.sorting[0].name }}
```

File: src/store/modules/files.js [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+    const outputFileTypes = await client.getOutputFileTypes()
+    commit('SET_OUTPUT_FILE_TYPES', outputFileTypes)
+  }
+}
+
+const mutations = {
+  [RESET_ALL](state) {
+    const isDarkTheme = state.isDarkTheme
+    Object.assign(state, { ...initialState })
+    state.isDarkTheme = isDarkTheme
```
Comment (NicoPennec) [inline src/store/modules/files.js:35]: why `isDarkTheme` is useful here?

File: src/store/modules/files.js [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+  fileStatusMap: new Map(),
+  outputFileTypes: [],
+  outputFileTypeMap: new Map()
+}
+
+const state = { ...initialState }
+
+const getters = {
+  fileStatusMap: state => state.fileStatusMap,
+  outputFileTypeMap: state => state.outputFileTypeMap
```
Comment (NicoPennec) [inline src/store/modules/files.js:16]: this map is never set.

File: src/store/modules/files.js [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+const actions = {
+  async loadFileStatuses({ commit }) {
+    const fileStatuses = await client.getFileStatuses()
+    commit('SET_FILE_STATUSES', fileStatuses)
```
Comment (NicoPennec) [inline src/store/modules/files.js:22]: This mutation should be declared an import through `mutation-types.js`

File: src/store/modules/files.js
Comment (NicoPennec) [inline src/store/modules/files.js:1]: this store file seems to be on WIP state.

Comment (frankrousseau) [reply src/store/modules/files.js:1]: Yes, this is what I wrote in the commits.

File: src/components/tops/ActionPanel.vue (hunk 14 above)
Comment (frankrousseau) [reply src/components/tops/ActionPanel.vue:1840]: No, `border-color` leads often to unexpected behaviour.

File: src/components/previews/PreviewPlayer.vue (hunk 11 above)
Comment (frankrousseau) [reply src/components/previews/PreviewPlayer.vue:1561]: It's not possible because we need to clone the annotation.

**Comment by frankrousseau** [comment]
Thx @NicoPennec for the review

File: src/components/previews/PreviewPlayer.vue (hunk 11 above)
Comment (NicoPennec) [reply src/components/previews/PreviewPlayer.vue:1562]: But `filter()` create a shallow copy similar to the spread operator `{... a}`
